packages/cactus-plugin-ledger-connector-iroha2/src/test/typescript/integration/iroha-instructions-and-queries.test.ts

```diff
@@ -0,0 +1,591 @@
+  Iroha2KeyPair,
+} from "../../../main/typescript/public-api";
+import {
+  IrohaV2TestEnv,
+  generateTestIrohaCredentials,
+  waitForCommit,
+} from "../test-helpers/iroha2-env-setup";
+import { addRandomSuffix } from "../test-helpers/utils";
+
+import { bytesToHex } from "hada";
+import "jest-extended";
+
+// Logger setup
+const log: Logger = LoggerProvider.getOrCreate({
+  label: "iroha-instructions-and-queries.test",
+  level: testLogLevel,
+});
+
+/**
+ * Main test suite
+ */
+describe("Instructions and Queries test", () => {
+  let env: IrohaV2TestEnv;
+
+  beforeAll(async () => {
+    env = new IrohaV2TestEnv(log);
+    await env.start();
+  });
+
+  afterAll(async () => {
+    if (env) {
+      await env.stop();
+    }
+  });
+
+  describe("Domain tests", () => {
+    let domainName: string;
+
+    // Create domain common test
+    beforeAll(async () => {
+      // Generate random domain and assets name
+      domainName = addRandomSuffix("funcTestDomain");
+      expect(domainName).toBeTruthy();
+
+      // Create new domain
+      const transactionResponse = await env.apiClient.transactV1({
+        instruction: {
+          name: IrohaInstruction.RegisterDomain,
+          params: [domainName],
+        },
+        baseConfig: env.defaultBaseConfig,
+      });
+      expect(transactionResponse).toBeTruthy();
+      expect(transactionResponse.status).toEqual(200);
+      expect(transactionResponse.data.status).toBeTruthy();
+      expect(transactionResponse.data.status).toEqual("OK");
+
+      // Sleep
+      await waitForCommit();
+    });
+
+    test("Query single domain (FindDomainById)", async () => {
+      const queryResponse = await env.apiClient.queryV1({
+        queryName: IrohaQuery.FindDomainById,
+        baseConfig: env.defaultBaseConfig,
+        params: [domainName],
+      });
+      expect(queryResponse).toBeTruthy();
+      expect(queryResponse.data).toBeTruthy();
+      expect(queryResponse.data.response).toBeTruthy();
+      expect(queryResponse.data.response.id).toBeTruthy();
+      expect(queryResponse.data.response.id.name).toEqual(domainName);
+    });
+
+    test("Query all domains (FindAllDomains)", async () => {
+      const queryResponse = await env.apiClient.queryV1({
+        queryName: IrohaQuery.FindAllDomains,
+        baseConfig: env.defaultBaseConfig,
+      });
+      expect(queryResponse).toBeTruthy();
+      expect(queryResponse.data).toBeTruthy();
+      expect(queryResponse.data.response).toBeTruthy();
+      expect(JSON.stringify(queryResponse.data.response)).toContain(domainName);
+    });
+  });
+
+  describe("Account tests", () => {
+    let newAccountName: string;
+    let newAccountDomainName: string;
+    let newAccountCredentials: Iroha2KeyPair;
+
+    // Register new account (RegisterAccount)
+    beforeAll(async () => {
+      newAccountName = addRandomSuffix("fooAcc");
+      expect(newAccountName).toBeTruthy();
+      newAccountDomainName = addRandomSuffix("newAccDomain");
```

Comment:
@petermetz These has been resolved. I kept distinctive names with random suffix for easier debugging of tests.

Reply:
@outSH Got it, thanks. I'm dismissing the alert because Math.random is bad, but it's fine for test cases IMO since that Math.random will never run in production code.